src/cloudformation/structure/cloudformation_parser.go

```diff
@@ -54,6 +57,41 @@ func (p *CloudformationParser) GetSupportedFileExtensions() []string {
 	return []string{common.YamlFileType.Extension, common.YmlFileType.Extension, common.CFTFileType.Extension, common.JSONFileType.Extension}
 }
 
+// Validate file has AWSTemplateFormatVersion
+func (p *CloudformationParser) ValidFile(filePath string) bool {
+	// #nosec G304
+	file, err := os.Open(filePath)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("Error opening file %s, skipping: %v", filePath, err))
+		return false
+	}
+	bytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("Error reading file %s, skipping: %v", filePath, err))
+		return false
+	}
+	if err = file.Close(); err != nil {
+		logger.Warning(fmt.Sprintf("Error closing file %s, skipping: %v", filePath, err))
+		return false
+	}
+
+	if !strings.HasSuffix(filePath, ".json") {
```

Comment:
@alexjurkiewicz A common suffix is `.template`, for both json and yaml templates - it's hard to lean on it. Might be better to leverage the existing `GetFileFormat` function

Reply:
I copied this check & "convert everything to JSON" logic from `goformation`. Since YAML is a superset of JSON, it's a no-op to run this conversion on a JSON file. I guess the suffix check is just a minor performance optimisation.

Reply:
Does that mean that in the next line, `sanathyaml.YAMLToJSON(bytes)`, if the bytes represent a json string, it won't break?

Reply:
that's correct. YAML is a superset of JSON, so you can rename a JSON file to YAML and it is valid